### PR TITLE
Create type for DeploymentParams (PR 8)

### LINKED ISSUE #316: Create type for deploymentParameters
We don't have a type for `ExecutionEngineState.deploymentParameters`.

image

The key should be named moduleId. This is repeated 9 times in the codebase. We should find where we are defining it ad-hoc and use that type.

### PR DESCRIPTION
**Pull request by zoeyTM**
fixes #316

### CODE REVIEW

File: packages/core/src/new-api/internal/validation/validate.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ import { validateSendData } from "./futures/validateSendData";
 export async function validate(
   module: IgnitionModule,
   artifactLoader: ArtifactResolver,
-  moduleParameters: { [key: string]: ModuleParameters }
+  moduleParameters: DeploymentParameters
```
Comment (kanej) [inline packages/core/src/new-api/internal/validation/validate.ts:23]: Lets rename `moduleParameters` to `deploymentParameters` here.

File: packages/core/src/new-api/types/module.ts [hunk 2]
```diff
@@ -361,6 +361,15 @@ export interface ModuleParameters {
   [parameterName: string]: ModuleParameterType;
 }
 
+/**
+ * An object containing a map of module ID's to their respective ModuleParameters.
+ *
+ * @beta
+ */
+export interface DeploymentParameters {
```
Comment (kanej) [inline packages/core/src/new-api/types/module.ts:369]: I think we should move this to `./types/deployer.ts`, as that is where the user will encounter it first.

What do you think?

**Review by kanej — APPROVED** [review]
I left a question about the best place to put the types, but after whatever the decision is there it is good to merge.

File: packages/core/src/new-api/types/module.ts (hunk 2 above)
Comment (alcuadrado) [reply packages/core/src/new-api/types/module.ts:369]: Makes sense to me

**Pull request merged**